models/brac/function/get_dashboard_data_anc_impact.sql

```diff
@@ -0,0 +1,538 @@
+			
+			GROUP BY
+				preg.uuid,
+				preg.patient_id,
+				preg.reported_by_parent,
+				period_reported,
+				period_2nd_tri_end,
+				period_mdd,
+				preg.early_reg,
+				preg.danger_sign_at_reg,
+				preg.has_lmp
+				
+				
+		)
+		
+--######################
+--MAIN QUERY STARTS HERE
+--######################
+
+/*parameterized group by*/
+/*this version is specifically for LG config where 'clinic' shouldn't be a group by category...lowest level is health_center' */
+SELECT
+
+	CASE
+		WHEN param_facility_group_by = 'clinic' OR param_facility_group_by = 'health_center' OR param_facility_group_by = 'district_hospital'
+		THEN place_period.district_hospital_uuid
+		ELSE 'All'
+	END AS _district_hospital_uuid,	
+	CASE
+		WHEN param_facility_group_by = 'clinic' OR param_facility_group_by = 'health_center' OR param_facility_group_by = 'district_hospital'
+		THEN place_period.district_hospital_name
+		ELSE 'All'
+	END AS _district_hospital_name,
+	
+	CASE
+		WHEN param_facility_group_by = 'clinic' OR param_facility_group_by = 'health_center'
+		THEN place_period.health_center_uuid
+		ELSE 'All'
+	END AS _health_center_uuid,			
+	CASE
+		WHEN param_facility_group_by = 'clinic' OR param_facility_group_by = 'health_center'
+		THEN place_period.health_center_name
+		ELSE 'All'
+	END AS _health_center_name,
+		
+--	CASE
+--		WHEN param_facility_group_by = 'clinic'
+--		THEN place_period.clinic_uuid
+--		ELSE 'All'
+--	END AS _clinic_uuid,
+--	CASE
+--		WHEN param_facility_group_by = 'clinic'
+--		THEN place_period.clinic_name
+--		ELSE 'All'
+--	END AS _clinic_name,
+	
+	'clinic_uuid'::text AS _clinic_uuid,
+	'clinic_name'::text AS _clinic_name,
+	
+	place_period.period_start AS _period_start,
+	date_part('epoch',place_period.period_start)::numeric AS _period_start_epoch,
+	
+	CASE
+--		WHEN param_facility_group_by = 'clinic'
+--			THEN place_period.clinic_uuid
+		WHEN param_facility_group_by = 'health_center'
+			THEN place_period.health_center_uuid
+		WHEN param_facility_group_by = 'district_hospital'
+			THEN place_period.district_hospital_uuid
+		ELSE
+			'All'
+	END AS _facility_join_field,
+		
+		/* By Reg Date */
+		SUM(COALESCE(count_new_person_by_reg,0)) AS count_new_person_by_reg,
+		SUM(COALESCE(count_preg_by_reg,0)) AS count_preg_by_reg,
+		SUM(COALESCE(count_preg_with_lmp_by_reg,0)) AS count_preg_with_lmp_by_reg,
+		SUM(COALESCE(count_preg_without_lmp_by_reg,0)) AS count_preg_without_lmp_by_reg,
+		
+		
+		SUM(COALESCE(count_preg_early_reg_by_reg,0)) AS count_preg_early_reg_by_reg,
+		SUM(COALESCE(count_deliv_by_reg,0)) AS count_deliv_by_reg,
+		SUM(COALESCE(count_danger_sign_by_reg,0)) AS count_danger_sign_by_reg,
+		SUM(COALESCE(count_pregnancy_visit_by_reg,0)) AS count_pregnancy_visit_by_reg,
+	
+		CASE
+			WHEN SUM(COALESCE(count_preg_by_reg,0)) = 0
+			THEN 0::float
+			ELSE (SUM(COALESCE(count_preg_early_reg_by_reg,0))::float / SUM(COALESCE(count_preg_by_reg,0))::float) * 100            
```

Comment:
IIRC, there exists a safe_divide user-defined function that can be used for this and similar instances